### ISSUE TITLE
Renaming

### ISSUE THREAD

**Issue by uxkjaer** [marcellourbani/vscode_abap_remote_fs#154]
The /sap/bc/adt/refactoring API allows the remaing.

Example /sap/bc/adt/refactorings?step=evaluate&rel=http%3A%2F%2Fwww.sap.com%2Fadt%2Frelations%2Frefactoring%2Frename&uri=%2Fsap%2Fbc%2Fadt%2Fprograms%2Fprograms%2Fztest%2Fsource%2Fmain%23start%3D19%2C12%3Bend%3D19%2C12

Sends back response
`<?xml version="1.0" encoding="UTF-8"?><rename:renameRefactoring xmlns:rename="http://www.sap.com/adt/refactoring/renamerefactoring">
  <rename:oldName>test</rename:oldName>
  <rename:newName>test</rename:newName>
  <generic:genericRefactoring xmlns:generic="http://www.sap.com/adt/refactoring/genericrefactoring">
    <generic:title>Rename Method</generic:title>
    <generic:adtObjectUri>/sap/bc/adt/programs/programs/ztest/source/main#start=19,12;end=19,12</generic:adtObjectUri>
    <generic:affectedObjects>
      <generic:affectedObject xmlns:adtcore="http://www.sap.com/adt/core" adtcore:uri="/sap/bc/adt/programs/programs/ztest/source/main" adtcore:type="PROG/P" adtcore:name="ZTEST">
        <generic:textReplaceDeltas>
          <generic:textReplaceDelta>
            <generic:rangeFragment>#start=14,8;end=14,12</generic:rangeFragment>
            <generic:contentOld/>
            <generic:contentNew/>
          </generic:textReplaceDelta>
          <generic:textReplaceDelta>
            <generic:rangeFragment>#start=19,9;end=19,13</generic:rangeFragment>
            <generic:contentOld/>
            <generic:contentNew/>
          </generic:textReplaceDelta>
        </generic:textReplaceDeltas>
        <generic:userContent>25-413-D203075401E94CB2D7E175BCE1FE1D3D64D068BF</generic:userContent>
      </generic:affectedObject>
    </generic:affectedObjects>
    <generic:transport/>
    <generic:ignoreSyntaxErrorsAllowed>true</generic:ignoreSyntaxErrorsAllowed>
    <generic:ignoreSyntaxErrors>false</generic:ignoreSyntaxErrors>
    <generic:userContent/>
  </generic:genericRefactoring>
  <rename:userContent/>
</rename:renameRefactoring>
`

Add the generic:contentOld and generic:contentNew and this will do the renaming.

Currently only tested on local program. Will be alot more complex with transports and other packages involved I assume.

**Comment by marcellourbani** [comment]
Yep that would be good to add. Don't remember if I have support in the API yet.
Might be tricky to integrate in vscode, but the main reason why I didn't do it when adding quickfixes (not being able to deal with multiple "files") is gone
